sale_coupon_multi_use/models/sale_order_line.py

```diff
@@ -19,6 +19,6 @@ def unlink(self):
         consumption_lines = self.mapped("coupon_consumption_line_ids")
         if consumption_lines:
             # Remove related consumption lines
-            # Using sudo because no user can remove consumption lines
-            consumption_lines.sudo().unlink()
+            # Using super user because no user can remove consumption lines
+            consumption_lines.with_user(SUPERUSER_ID).unlink()
```

Comment:
Actually, sudo() still exists and apply automatically SUPERUSER_ID. But with_user() allows to 'do the operations with high privileges keeping the user details - logging for instance'.

sudo : https://github.com/odoo/odoo/blob/13.0/odoo/models.py#L5150

with_user: https://github.com/odoo/odoo/blob/13.0/odoo/models.py#L5161


So, you can let sudo() here :smiley:

Reply:
Now it's done: it will be more easy to migrate in future version ;)